compiler/noirc_evaluator/src/ssa/opt/remove_bit_shifts.rs

```diff
@@ -224,6 +221,10 @@ impl Context<'_, '_, '_> {
                     BinaryOp::Sub { unchecked: true },
                     lhs_sign_as_int,
                 );
+
+                // Similar to the above wrapping addition, this is a wrapping subtraction because if
+                // `shifted_complement` is `i8 0`, so `Field 0` which, so the subtraction would give
+                // the maximum Field value which, when truncated, equals `i8 -1`.
```

Comment:
I don't think this is correct. It's commented above that this cannot overflow so we're just using an unchecked operation to avoid an unnecessary range check.

 If `shifted_complement == 0` and `lhs_sign_as_int == 1` as required for this to overflow then we'd get `0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000000` as the result which would not truncate into a signed type correctly.